### PR TITLE
Plugin Docs

### PR DESCRIPTION
**Pull request by coffee-cup**
- Small changes to readme
- add docs for plugins
- docs typo fixes
- add footer

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+Fork, then clone the repo:
+
+```
+git clone https://github.com/your-username/prodo.git
+```
+
+#### Building
+
+Build all packages and examples with.
```
Comment (andrejak) [inline CONTRIBUTING.md]:
```suggestion
Build all packages and examples with:
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+```
+yarn build
+```
+
+This typechecks all TypeScript and converts it to JavaScript. When developing we
+recommend building the project in watch mode
```
Comment (andrejak) [inline CONTRIBUTING.md]:
```suggestion
recommend building the project in watch mode:
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+yarn verify
+```
+
+Only run typechecking and linting:
+
+```
+yarn lint
+```
+
+To apply suggested linting fixes
```
Comment (andrejak) [inline CONTRIBUTING.md]:
```suggestion
To apply suggested linting fixes:
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+```
+yarn lint --fix
+```
+
+Only run tests
```
Comment (andrejak) [inline CONTRIBUTING.md]:
```suggestion
Only run tests:
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+with a proposal for a new feature before starting on the work. However, we also
+welcome pull requests.
+
+The contribution workflow looks like
+
+- Open an new issue in the [issue
+  tracker](https://github.com/prodo-ai/prodo/issues/new)
+- Fork the repo
+- Create a new feature branch based off the `master` branch
+- Make sure all tests pass and there are not linting errors
```
Comment (andrejak) [inline CONTRIBUTING.md]:
```suggestion
- Make sure all tests pass and there are no linting errors
```

File: docs/content/03_plugins/effects.md [hunk 6]
```diff
@@ -2,3 +2,119 @@
...
+  expect(state).toEqual({
+    T1: {
+      text: "foo",
+      done: false,
+    },
+  });
+})
+```
+
+The problem with this is that we do not know what the new todos id will be,
```
Comment (andrejak) [inline docs/content/03_plugins/effects.md]:
```suggestion
The problem with this is that we do not know what the new todo's id will be,
```

File: docs/content/03_plugins/logger.md [hunk 7]
```diff
@@ -2,3 +2,44 @@
...
+
+```ts
+export interface LoggerConfig {
+  logger?: boolean;
+}
+```
+
+`logger`
+
+boolean that indicates whether or not logging is enabled. Defaults. to `false`.
```
Comment (andrejak) [inline docs/content/03_plugins/logger.md]:
```suggestion
boolean that indicates whether or not logging is enabled. Defaults to `false`.
```

File: docs/content/03_plugins/effects.md [hunk 8]
```diff
@@ -2,3 +2,119 @@
...
+    initState: { todos: {} },
+    mockEffects: {
+      randomId: ["T1"],
+    },
+  });
+
+  const { state } = await store.dispatch(newTodo)("buy milk");
+  expect(state).toEqual({
+    T1: {
+      text: "foo",
```
Comment (andrejak) [inline docs/content/03_plugins/effects.md]: Shouldn't text here be "buy milk" based on the action argument?

```suggestion
      text: "buy milk",
```

File: docs/content/03_plugins/effects.md [hunk 9]
```diff
@@ -2,3 +2,119 @@
...
+```ts
+test("create a new todo", async () => {
+  const { store } = model.createStore({
+    initState: { todos: {} },
+  });
+
+  const { state } = await store.dispatch(newTodo)("buy milk");
+  expect(state).toEqual({
+    T1: {
+      text: "foo",
```
Comment (andrejak) [inline docs/content/03_plugins/effects.md]: Shouldn't text here be "buy milk" based on the action argument?

```suggestion
      text: "buy milk",
```

File: docs/content/03_plugins/routing.md [hunk 10]
```diff
@@ -2,3 +2,345 @@
...
+`replace`
+
+The `replace` action is similar to the `push` action, except that it will replace the current entry in the history instead pushing a new entry. This means that you cannot use the browser's back button to return to the previous route.
+
+```ts
+import { replace } from "@prodo/route";
+
+dispatch(replace)("/a/b/c");
+// or
+dispatch(replcae)({
```
Comment (andrejak) [inline docs/content/03_plugins/routing.md]:
```suggestion
dispatch(replace)({
```

File: docs/content/03_plugins/routing.md [hunk 11]
```diff
@@ -2,3 +2,345 @@
...
+  path: "/a/b/c",
+});
+```
+
+### Declarative API
+
+To allow for more declarative routing, the plugin also provides a few React components that can be used to declarative determine what to render, and add elements such as links to your page. These are designed to be as close as possible to the counterparts defined in the popular [`react-router`](https://www.npmjs.com/package/react-router) library:
```
Comment (andrejak) [inline docs/content/03_plugins/routing.md]:
```suggestion
To allow for more declarative routing, the plugin also provides a few React components that can be used to declaratively determine what to render, and add elements such as links to your page. These are designed to be as close as possible to the counterparts defined in the popular [`react-router`](https://www.npmjs.com/package/react-router) library:
```

File: docs/content/03_plugins/effects.md (hunk 8 above)
Comment (coffee-cup) [reply docs/content/03_plugins/effects.md]: good catch

File: docs/content/04_advanced/creating-plugins.md [hunk 12]
```diff
@@ -143,17 +125,22 @@ variables or methods available to the user in their component. The `universe`
 **is not** directly modifiable and you should [use actions](#plugin-actions) if you want to modify it.
 
 There are two options available to subscribe the calling component to a
-particular path of the [universe](#universe). These are
+particular path of the [universe](./universe). These are
 
-- using `createUniverseWatcher`
-- calling `ctx.subscribe`
+- Using `createUniverseWatcher`.
```
Comment (coffee-cup) [inline docs/content/04_advanced/creating-plugins.md]:
```suggestion
- Using `createUniverseWatcher`
```

**Pull request merged**